## Hypothesis Testing

In [1]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [2]:
import numpy as np
import pandas as pd
from scipy.stats import chi2_contingency
from scipy.stats import ttest_ind

df = pd.read_csv('/content/drive/MyDrive/ImpactSense_Oct25/data/preprocessed_earthquake_data.csv')

#### 1. Magnitude Difference by Earthquake Type
Hypothesis: The mean earthquake magnitude differs significantly between reviewed and non-reviewed events.

In [3]:
# Select relevant groups, drop missing values
group1 = df[df['status'] == 'reviewed']['mag'].dropna()
group2 = df[df['status'] != 'reviewed']['mag'].dropna()

ttest_result = ttest_ind(group1, group2, equal_var=False)
print('t-test statistic:', ttest_result.statistic)
print('p-value:', ttest_result.pvalue)

t-test statistic: -1.0748565111189015
p-value: 0.28494966597262317


**Statement**:The t-test result for Hypothesis-1 yields a t-statistic of -1.07 and a p-value of 0.285. Since the p-value is greater than the common significance threshold of 0.05, there is not enough evidence to reject the null hypothesis. Thus, the mean earthquake magnitude does not differ significantly between reviewed and non-reviewed events in your data.

#### 2. Depth Difference by Magnitude Type
Hypothesis: The mean depth of earthquakes significantly varies between 'mb' and 'mw' magnitude type categories.

In [4]:
group1 = df[df['magType'] == 'mb']['depth'].dropna()
group2 = df[df['magType'] == 'mw']['depth'].dropna()

ttest_result = ttest_ind(group1, group2, equal_var=False)
print('t-test statistic:', ttest_result.statistic)
print('p-value:', ttest_result.pvalue)

t-test statistic: 22.514758657461467
p-value: 8.933021714439036e-112


**Statement**: The t-test result for this hypothesis yields a t-test statistic of 22.51 and a p-value of approximately $8.93 \times 10^{-112}$. Since this p-value is extremely small and far below the conventional significance threshold of 0.05, there is extremely strong evidence to reject the null hypothesis. Therefore, the mean depth of earthquakes is significantly different between `mb` and `mw` magnitude type events in your dataset.

#### 3. Magnitude Consistency Across Months
Hypothesis: There is no significant association between the month of occurrence and the categorical bins of earthquake magnitude (such as low, medium, high).

In [5]:
# Bin magnitudes into categories
df['mag_bin'] = pd.cut(df['mag'], bins=[0, 3.5, 5.0, 6.5, 9], labels=['Low', 'Medium', 'High', 'Very High'])

# Contingency table: mag_bin vs Month
contingency = pd.crosstab(df['mag_bin'], df['Month'])
chi2, p, dof, expected = chi2_contingency(contingency)
print('Chi-square statistic:', chi2)
print('Degrees of freedom:', dof)
print('p-value:', p)

Chi-square statistic: 46.542726462177434
Degrees of freedom: 33
p-value: 0.05922370884750974


**Statement**: The chi-square test result for this hypothesis yields a chi-square statistic of 46.54, degrees of freedom of 33, and a p-value of approximately 0.0592. Since this p-value is slightly above the conventional significance threshold of 0.05, there is insufficient evidence to reject the null hypothesis at the 5% level. Therefore, the data does not provide strong enough evidence to conclude that earthquake magnitude category and month of occurrence are dependent variables.

#### 4. DayOfWeek Consistency Across Event Type
Hypothesis: The type of event ('type' column: earthquake, quarry blast, etc.) is independent of the day of week
(DayOfWeek) on which it occurred.

In [6]:
contingency = pd.crosstab(df['type'], df['DayOfWeek'])
chi2, p, dof, expected = chi2_contingency(contingency)
print('Chi-square statistic:', chi2)
print('Degrees of freedom:', dof)
print('p-value:', p)

Chi-square statistic: 92.16992465854213
Degrees of freedom: 36
p-value: 8.07286408381138e-07


**Statement**: The chi-square test result for this hypothesis yields a chi-square statistic of 92.17, degrees of freedom of 36, and a p-value of approximately $8.072 \times 10^{-07}$
 . Since this p-value is far below the conventional significance threshold of 0.05, there is very strong evidence to reject the null hypothesis. Therefore, the day of the week and event type are significantly dependent, indicating a meaningful association between these categorical variables in your data.

#### 5. Magnitude Mean Difference by Year Decade
Hypothesis: The mean earthquake magnitude (mag) differs significantly between records from two different decades (e.g., 2000s vs 2010s).

In [7]:
df['decade'] = (df['Year'] // 10) * 10
dec1 = df[df['decade'] == 2000]['mag'].dropna()
dec2 = df[df['decade'] == 2010]['mag'].dropna()

ttest_result = ttest_ind(dec1, dec2, equal_var=False)
print('t-test statistic:', ttest_result.statistic)
print('p-value:', ttest_result.pvalue)

t-test statistic: 7.034358911416218
p-value: 2.0392846020245177e-12


**Statement**: The t-test result for this hypothesis yields a t-test statistic of 7.03 and a p-value of approximately $2.039 \times 10^{-12}$. Since this p-value is extremely small and far below the conventional significance threshold of 0.05, there is very strong evidence to reject the null hypothesis. Therefore, the mean earthquake magnitude differs significantly between the two compared decades in your dataset.

### Task:

- Suggest five additional hypothesis tests, based on the preprocessed earthquake dataset, that cover both numerical and categorical variables using t-tests and chi-square tests, to extend the depth of your analysis.

Earthquake Magnitude vs. Day vs. Night Occurrence

In [8]:
# Hypothesis: The mean earthquake magnitude differs significantly between day and night occurrences.

# Define day and night based on a time column (assuming 'Hour' exists)
# You may need to adjust the time definition based on your dataset and location
day_hours = list(range(6, 18)) # 6 AM to 6 PM

df['is_day'] = df['Hour'].apply(lambda x: x in day_hours)

# Separate magnitudes for day and night
day_magnitudes = df[df['is_day']]['mag'].dropna()
night_magnitudes = df[~df['is_day']]['mag'].dropna()

# Perform independent samples t-test
ttest_result = ttest_ind(day_magnitudes, night_magnitudes, equal_var=False)

print('t-test statistic:', ttest_result.statistic)
print('p-value:', ttest_result.pvalue)

t-test statistic: 0.6113528976528185
p-value: 0.5409672435972561


**Statement**: The t-test result for this hypothesis yields a t-test statistic of 0.611 and a p-value of approximately 0.541. Since this p-value is greater than the conventional significance threshold of 0.05, there is insufficient evidence to reject the null hypothesis. Therefore, the data does not provide strong enough evidence to conclude that the mean earthquake magnitude differs significantly between day and night occurrences in your dataset.

Depth Differences Between Two Geographical Regions

In [9]:
# Hypothesis: The mean depth of earthquakes differs significantly between two different geographical regions.

# Define two geographical regions based on latitude and longitude (these are example boundaries)
# You will likely need to adjust these based on the regions you want to compare
region1_lat_min, region1_lat_max = 30, 40
region1_lon_min, region1_lon_max = -120, -110

region2_lat_min, region2_lat_max = -30, -20
region2_lon_min, region2_lon_max = 170, 180

# Select depth data for each region, dropping missing values
region1_depths = df[
    (df['latitude'] >= region1_lat_min) & (df['latitude'] <= region1_lat_max) &
    (df['longitude'] >= region1_lon_min) & (df['longitude'] <= region1_lon_max)
]['depth'].dropna()

region2_depths = df[
    (df['latitude'] >= region2_lat_min) & (df['latitude'] <= region2_lat_max) &
    (df['longitude'] >= region2_lon_min) & (df['longitude'] <= region2_lon_max)
]['depth'].dropna()

# Check if there are enough samples in each region
if len(region1_depths) < 2 or len(region2_depths) < 2:
    print("Not enough data in one or both regions to perform a t-test.")
else:
    # Perform independent samples t-test
    ttest_result = ttest_ind(region1_depths, region2_depths, equal_var=False)

    print('t-test statistic:', ttest_result.statistic)
    print('p-value:', ttest_result.pvalue)

t-test statistic: -32.3999580874509
p-value: 1.2397462023823101e-180


**Statement**: The t-test result for this hypothesis yields a t-test statistic of -32.40 and a p-value of approximately $1.24 \times 10^{-180}$. Since this p-value is extremely small and far below the conventional significance threshold of 0.05, there is very strong evidence to reject the null hypothesis. Therefore, the mean earthquake depth differs significantly between the two compared geographical regions in your dataset.

Earthquake Frequency by Month

In [13]:
# Hypothesis: The frequency of earthquakes is independent of the month of occurrence.

# Create a contingency table of earthquake counts by month
contingency = pd.crosstab(df['Month'], columns='count')

# Perform chi-square test of independence
chi2, p, dof, expected = chi2_contingency(contingency)

print('Chi-square statistic:', chi2)
print('Degrees of freedom:', dof)
print('p-value:', p)

Chi-square statistic: 0.0
Degrees of freedom: 0
p-value: 1.0


**Statement**: The chi-square test result for this hypothesis yields a chi-square statistic of 0.0, degrees of freedom of 0, and a p-value of 1.0. A p-value of 1.0 indicates there is no evidence to reject the null hypothesis. However, a chi-square statistic of 0 and degrees of freedom of 0 suggest that the contingency table likely has only one row or column, which means there's only one category being compared (in this case, all data is in one 'Month' category in the contingency table as it was created). This test is not meaningful as it is currently set up. To properly test this hypothesis, the contingency table should have multiple months to compare frequencies across.